packages/flutter_tools/test/integration.shard/flutter_plugins_dependencies_test.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'package:file/file.dart';
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+void main() {
+  late Directory tempDir;
+
+  setUp(() async {
+    tempDir =
+        createResolvedTempDirectorySync('flutter_plugins_dependencies_test.');
+  });
+
+  tearDown(() async {
+    tryToDelete(tempDir);
+  });
+
+  test(
+      '.flutter-plugins-dependencies correctly denotes project dev dependencies on iOS, Android',
+      () async {
+    // Create Flutter project.
+    final String flutterBin =
+        fileSystem.path.join(getFlutterRoot(), 'bin', 'flutter');
+
+    await processManager.run(<String>[
+      flutterBin,
+      'create',
+      tempDir.path,
+      '--project-name=testapp',
+    ], workingDirectory: tempDir.path);
+
+    final File pubspecFile = tempDir.childFile('pubspec.yaml');
+    expect(pubspecFile.existsSync(), true);
+    String pubspecFileAsString = pubspecFile.readAsStringSync();
```

Comment:
This might be over-testing for an integration test or fragile if the templates change. I might consider instead shelling out and using `flutter pub add`, i.e. `flutter pub add camera:0.11.0`.

The other problem though is I'd be worried that eventually when the current Flutter SDK moves past these versions, this will cause a breakage (cannot resolve X for Y).

A potentially cleaner, but slightly more cumbersome path might be to have the integration test create two (blank) plugins instead, i.e. a `plugin_a_real_dependency` and `plugin_b_dev_dependency`; then we could use `path: ...` dependencies and not worry about version solving in this integration test.

(Let me know if you could use more of an explanation or help doing this)

Reply:
Gotcha this approach simplifies things a lot so thank you! I'll make this change and then see if I'm still having trouble getting the plugins marked as dev dependencies.